Tools/DumpRenderTree/mac/LayoutTestHelper.m

```diff
@@ -242,15 +250,16 @@ static void lockDownDiscreteGraphics(void)
     static io_connect_t permanentLockDownService = 0;
 
     // This call stalls until the graphics device lock is granted.
+    kern_return_t kernResult;
     kernResult = IOServiceOpen(serviceObject, mach_task_self(), 1, &permanentLockDownService);
     if (kernResult != KERN_SUCCESS) {
         NSLog(@"IOServiceOpen() failed in %s with kernResult = %d", __FUNCTION__, kernResult);
         return;
     }
 
-    kernResult = IOObjectRelease(serviceObject);
+    kernResult = IOServiceClose(serviceObject);
```

Comment:
Ordinarily this would be the right thing to do. But you missed the comment on line 240:

```
// We're intentionally leaking this io_connect in order for the process to stay locked to discrete graphics
// for the lifetime of the service connection.
```

By calling `IOObjectRelease` we release the bulk of the object memory, while holding the connection open for the life of the process. If you call `IOServiceClose`, it will shut down that connection removing the hold on the discrete GPU.